add support for Ubuntu 20.04

Octav14n opened the pull request:
I added the instructions and changes I needed to make to make the program runable on Ubuntu 20.04.

g++ seems to need the input-file and output-parameter before the includes are declared (why ever that is?!)
Ubuntu 20.04 does not have the needed packages in its repository so I added a section in README on how to install the dependencies.

Octav14n committed:
Add dependency install instructions.

Octav14n committed:
Change g++ parameter order

fixes ld errors on my machine (ubuntu 20.04)

recolic commented on tools/list_keyrings.sh:
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
-eval g++ $(pkg-config --cflags --libs gnome-keyring-1) list_keyrings.cc -o list.out &&
+cd $(dirname $0)
```
Please write `cd $(dirname $0) &&`. Any small error should fail the whole script, as early as possible.

recolic commented on README.md:
```diff
@@ -19,7 +19,20 @@ I encrypt the `keyring-name : password` pair with GnuPG and save it as `secret-f
 > I recommend you to **configure Yubikey as GPG smartcard**. The system would just ask you to unlock gnome-keyring with your default GPG software. You may generate a new GPG key for yubikey, or move your existing GPG key into yubikey. Refer to google for these knowledge. 
 
-First, build the project from source.
+First, install packages.
```
Not all users are using ubuntu. Please Add a new section named `dependency`, and add an entry `ubuntu 20.04`. Archlinux/Gentoo/fedora/opensuse users may use different packages.

Octav14n committed:
add exit-code check to list_keyrings.sh

Octav14n committed:
Extract dependencies in README

extract dependencies to new headline. At the moment I have only added how to install dependencies for Ubuntu 20.04.

Octav14n committed:
Re-Insert accidentially deleted line.

Octav14n commented:
OK, I made the suggested changes.
I found a line I accidentally deleted (I have no idea how/why) and re-inserted it too.

recolic commented:
Thanks!
